ui/app/components/ui/CodeSnippet.tsx

```diff
@@ -0,0 +1,167 @@
+import { Card, CardContent, CardHeader, CardTitle } from "~/components/ui/card";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "~/components/ui/tabs";
+import { Badge } from "~/components/ui/badge";
+import { Button } from "~/components/ui/button";
+import { Copy } from "lucide-react";
+
+export interface CodeTab {
+  id: string;
+  label: string;
+  content: string;
+  language?: string;
+  showLineNumbers?: boolean;
+}
+
+export interface CodeSnippetProps {
+  heading?: string;
+  tabs?: CodeTab[];
+  label?: string;
+  content?: string;
+  language?: string;
+  className?: string;
+  children?: React.ReactNode;
+  showLineNumbers?: boolean;
+}
+
+export interface CodeSnippetContentProps {
+  label?: string;
+  content?: string;
+  language?: string;
+  showLineNumbers?: boolean;
+}
+
+export function CodeSnippetHeader({ heading }: { heading?: string }) {
+  if (!heading) return null;
+
+  return (
+    <CardHeader className="pb-0">
+      <CardTitle className="text-xl">{heading}</CardTitle>
+    </CardHeader>
+  );
+}
+
+export function CodeSnippetTabs({
+  tabs,
+  defaultTab,
+}: {
+  tabs?: CodeTab[];
+  defaultTab?: string;
+}) {
+  if (!tabs || tabs.length === 0) return null;
+
+  const defaultTabId = defaultTab || tabs[0].id;
+
+  return (
+    <Tabs defaultValue={defaultTabId} className="w-full">
+      <TabsList className="mb-2">
+        {tabs.map((tab) => (
+          <TabsTrigger key={tab.id} value={tab.id}>
+            {tab.label}
+          </TabsTrigger>
+        ))}
+      </TabsList>
+
+      {tabs.map((tab) => (
+        <TabsContent key={tab.id} value={tab.id}>
+          <CodeSnippetContent
+            label={tab.label}
+            content={tab.content}
+            language={tab.language}
+            showLineNumbers={tab.showLineNumbers}
+          />
+        </TabsContent>
+      ))}
+    </Tabs>
+  );
+}
+
+export function CodeSnippetContent({
+  label,
+  content,
+  showLineNumbers = false,
+}: CodeSnippetContentProps) {
+  if (!content) return null;
+
+  const handleCopy = () => {
+    navigator.clipboard.writeText(content);
+  };
```

Comment:
@maxbokov we just had an issue using this function since it only works in secure contexts. see [this PR](https://github.com/tensorzero/tensorzero/pull/1174) for your reference. I can fix this here.